libkbfs/tlf_edit_history.go

```diff
@@ -124,21 +155,57 @@ type TlfEditHistory struct {
 	fbo    *folderBranchOps
 	log    logger.Logger
 
-	lock      sync.Mutex
-	edits     TlfWriterEdits
-	editsTime time.Time
+	rmdsChan chan []ImmutableRootMetadata
+	wg       RepeatedWaitGroup
+
+	lock     sync.Mutex
+	edits    TlfWriterEdits
+	cancel   context.CancelFunc
+	shutdown bool
+	sends    sync.WaitGroup
+}
+
+// NewTlfEditHistory makes a new TLF edit history.
+func NewTlfEditHistory(config Config, fbo *folderBranchOps,
+	log logger.Logger) *TlfEditHistory {
+	teh := &TlfEditHistory{
+		config:   config,
+		fbo:      fbo,
+		log:      log,
+		rmdsChan: make(chan []ImmutableRootMetadata, 100),
```

Comment:
I can do this without storing `ctx`; I can just pass it directly to `process`.  Will do.